packages/ethereum-types/src/index.ts

```diff
@@ -6,10 +6,74 @@ export type JSONRPCErrorCallback = (err: Error | null, result?: JSONRPCResponseP
  * Do not create your own provider. Use an existing provider from a Web3 or ProviderEngine library
  * Read more about Providers in the 0x wiki.
  */
+export type SupportedProvider = Web3JsProvider | GanacheProvider | EIP1193Provider | ZeroExProvider;
+
+export type Web3JsProvider = Web3JsV1Provider | Web3JsV2Provider | Web3JsV3Provider;
+
+export interface GanacheProvider {
+    sendAsync(payload: JSONRPCRequestPayload, callback: JSONRPCErrorCallback): void;
+}
+
+// DEPRECATED(fabio): This interface should be replaced with the EIP 1193 provider interface
+// We will leave it here until the ecosystem has migrated fully to the new standard
 export interface Provider {
     sendAsync(payload: JSONRPCRequestPayload, callback: JSONRPCErrorCallback): void;
 }
 
+/**
+ * The interface for the provider used internally by 0x libraries
+ * Any property we use from any SupportedProvider should we explicitly
+ * add here
+ */
+export interface ZeroExProvider {
+    isZeroExProvider?: boolean;
+    isMetaMask?: boolean;
+    isParity?: boolean;
```

Comment:
Can we have a TODO to turn this remove this and use a getProviderType utility function which returns an enum. Feels a bit weird to just have these two (not Mist/Cipher/CBW etc)

Reply:
Can't easily do Cipher, CBW checks in Node.js. Added TODO.